pkg/util/nfacct/nfacct.go

```diff
@@ -0,0 +1,164 @@
+//go:build linux
+// +build linux
+
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfacct
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	utilexec "k8s.io/utils/exec"
+)
+
+const (
+	// nfacct binary
+	nfacctBinary = "nfacct"
```

Comment:
This is going to introduce a new external binary dependency, which is problematic because we can't assume all upgrading users will have it.

But it looks like we're only using it in a very simple way... we should figure out if we could plausibly do this with netlink rather than exec. (We already depend on a netlink library, `"github.com/vishvananda/netlink"`, though it doesn't seem to have any netfilter-related APIs besides conntrack, so you'd probably have to use the low-level bits, which are... pretty low-level. It may or may not be a good idea.)

If we can't do this with netlink then we'll have to think about whether this approach is better than the iptables-save-parsing approach...

Reply:
Will try this with netlink then

Reply:
Tried with netlink, was able to interact with the nfacct subsystem.

> so you'd probably have to use the low-level bits, which are... pretty low-level. It may or may not be a good idea.)

Yes, this is pretty low-level. Even if we don't pursue this, I won't have any regrets because the experience of working on it was enjoyable and educational.

Add/Get commands are working perfectly, List currently lists and flushes the counters.

Reply:
golang nfacct library? 😄